src/Client.php

```diff
@@ -111,14 +114,12 @@ class Client extends Magic
      * @param $baseUrl optional parameter for specifying the base URL when not using the default one
      *         (https://api.stormpath.com/v1).
      */
-    public function __construct(ApiKey $apiKey, $cacheManager, $cacheManagerOptions, $baseUrl = null, RequestSigner $requestSigner = null)
+    public function __construct(ApiKey $apiKey, $cacheManager, $cacheManagerOptions, $baseUrl = null, $authenticationScheme = Stormpath::SAUTHC1_AUTHENTICATION_SCHEME, HttpClient $httpClient = null, MessageFactory $messageFactory = null, UriFactory $uriFactory = null)
```

Comment:
It is, yes, but I fail to see when a any user to ever use this parameter?

If we wish to really preserve BC, we can certainly keep the parameter (and those RequestSigner classes), and do an instanceof to pick the authentication scheme. It's quite ugly though. If we want to keep superficial BC, we can just add a dummy parameter there.